src/java/za/co/mip/ablduck/ABLDuck.java

```diff
@@ -0,0 +1,417 @@
+                search.name = member.name;
+                search.icon = "icon-" + member.tagname;
+                search.url = "#!/api/" + member.owner + "-method-" + member.name;
+                search.sort = 3;
+                search.meta = member.meta;
+    
+                dataJSObject.search.add(search);
+            }
+            
+            //Hierarchy
+            HierarchyResult result = new HierarchyResult();
+            result = determineClassHierarchy(js, result);
+
+            List<String> hierarchy = result.getHierarchy();
+            Collections.reverse(hierarchy);
+            js.superclasses.addAll(hierarchy);
+
+            js.members.addAll(result.getInheritedmembers());
+
+            //Subclasses
+            for (Map.Entry<String, SourceJSObject> subclass:jsObjects.entrySet()) {
+                SourceJSObject subc = subclass.getValue();
+                if(subc.ext.equals(js.name)) 
+                    js.subclasses.add(subc.name);
+            }
+
+        }
+
+        //Write class js files out
+        for (Map.Entry<String, SourceJSObject> j:jsObjects.entrySet()) {
+            SourceJSObject js = j.getValue();
+
+            //Generate html
+            js.html = html.getClassHtml(jsObjects, js);
+    
+            File outputFile = new File(this.destDirOutput, js.name + ".js");
+            try (FileWriter file = new FileWriter(outputFile.toString())) {
+                file.write("Ext.data.JsonP." + js.name.replace(".", "_") + "(" + gson.toJson(js) + ");");
+            } catch (IOException ex) {
+                throw new BuildException(ex);
+            }
+        }
+
+        File dataFile = new File(this.destDir, "data.js");
+        try (FileWriter file = new FileWriter(dataFile.toString())) {
+            file.write("Docs = {\"data\":" + gson.toJson(dataJSObject) + "}");
+        } catch (IOException ex) {
+            throw new BuildException(ex);
+        }
+        
+    }
+
+    private HierarchyResult determineClassHierarchy(SourceJSObject curClass, HierarchyResult result) {
+        
+        String inherits = curClass.ext;
+
+        if (!"".equals(inherits)) {
+            result.addHierarchy(inherits);
+
+            SourceJSObject nextClass = jsObjects.get(inherits);
+                        
+            if (nextClass != null) {
+                for (MemberObject member : nextClass.members) {
+                    if (member.owner.equals(inherits) && (member.meta.isPrivate == null || !member.meta.isPrivate)) 
+                        result.addInheritedmember(member);
+                }
+            
+                determineClassHierarchy(nextClass, result);
+            }
+        }
+        return result;
+    }
+
+    private void extractTemplateDirectory(File outputDir) throws IOException {
+ 
+        File template = exportResource("ablduck.zip", outputDir);
+
+        unzip(template.getAbsolutePath(), outputDir.getAbsolutePath());
+
+        template.delete();

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "delete". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS899)